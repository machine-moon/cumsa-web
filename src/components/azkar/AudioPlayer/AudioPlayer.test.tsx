import React from "react";
import { render } from "@testing-library/react";
import { AudioPlayer } from "./index";

describe("AudioPlayer", () => {
  const defaultProps = {
    id: "test-audio",
    audioUrl: "https://example.com/audio.mp3",
    onPlay: jest.fn(),
    isActive: false,
    loading: false,
  };

  it("renders correctly when inactive", () => {
    const { container } = render(<AudioPlayer {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when active", () => {
    const { container } = render(<AudioPlayer {...defaultProps} isActive={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when loading", () => {
    const { container } = render(<AudioPlayer {...defaultProps} isActive={true} loading={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with label", () => {
    const { container } = render(<AudioPlayer {...defaultProps} label="Audio Label" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with error", () => {
    const { container } = render(
      <AudioPlayer {...defaultProps} isActive={true} error="Audio failed to load" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
