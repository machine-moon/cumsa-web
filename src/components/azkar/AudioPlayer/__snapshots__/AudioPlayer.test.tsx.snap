// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer renders correctly when active 1`] = `
<div
  class="flex items-center gap-2"
>
  <button
    class="min-w-[64px] text-center px-3 py-1 rounded-md text-xs font-medium transition-colors border shadow-sm focus:outline-none focus:ring-2 focus:ring-[var(--blue)]/50 bg-[var(--blue)] border-[var(--blue)] text-white"
    type="button"
  >
    Pause
  </button>
</div>
`;

exports[`AudioPlayer renders correctly when inactive 1`] = `
<div
  class="flex items-center gap-2"
>
  <button
    class="min-w-[64px] text-center px-3 py-1 rounded-md text-xs font-medium transition-colors border shadow-sm focus:outline-none focus:ring-2 focus:ring-[var(--blue)]/50 bg-white border-[var(--blue)]/30 text-[var(--foreground)] hover:bg-[var(--blue)] hover:text-white"
    type="button"
  >
    Play
  </button>
</div>
`;

exports[`AudioPlayer renders correctly when loading 1`] = `
<div
  class="flex items-center gap-2"
>
  <button
    class="min-w-[64px] text-center px-3 py-1 rounded-md text-xs font-medium transition-colors border shadow-sm focus:outline-none focus:ring-2 focus:ring-[var(--blue)]/50 bg-[var(--blue)] border-[var(--blue)] text-white"
    disabled=""
    type="button"
  >
    ...
  </button>
</div>
`;

exports[`AudioPlayer renders correctly with error 1`] = `
<div
  class="flex items-center gap-2"
>
  <button
    class="min-w-[64px] text-center px-3 py-1 rounded-md text-xs font-medium transition-colors border shadow-sm focus:outline-none focus:ring-2 focus:ring-[var(--blue)]/50 bg-[var(--blue)] border-[var(--blue)] text-white"
    type="button"
  >
    Pause
  </button>
  <span
    class="text-[10px] text-[var(--red)]"
  >
    Audio failed to load
  </span>
</div>
`;

exports[`AudioPlayer renders correctly with label 1`] = `
<div
  class="flex items-center gap-2"
>
  <button
    class="min-w-[64px] text-center px-3 py-1 rounded-md text-xs font-medium transition-colors border shadow-sm focus:outline-none focus:ring-2 focus:ring-[var(--blue)]/50 bg-white border-[var(--blue)]/30 text-[var(--foreground)] hover:bg-[var(--blue)] hover:text-white"
    type="button"
  >
    Play
  </button>
  <span
    class="text-[10px] text-[var(--muted)]"
  >
    Audio Label
  </span>
</div>
`;
